packages/amplify-cli/src/extensions/amplify-helpers/on-category-outputs-change.ts

```diff
@@ -63,3 +66,32 @@ function attachContextExtensions(context, packageLocation) {
     }
   }
 }
+
+// projects created before 5.2.0 didn't populate frontend config in amplify-meta.json
+// this method ensures frontend config settings are added to amplify meta on pull as they exist in parameters.json
+// https://app.asana.com/0/1200585422384147/1200740448709567/f
+function ensureAmplifyMetaFrontendConfig(context, amplifyMeta?) {
+  if (!amplifyMeta) {
+    amplifyMeta = stateManager.getMeta();
+  }
+
+  if (amplifyMeta.auth) {
+    const authResourceName = Object.keys(amplifyMeta.auth).find((key: any) => {
+      return amplifyMeta.auth[key].service === 'Cognito';
+    });
+
+    if (!authResourceName) return;
+
+    const authParameters: AuthParameters = stateManager.getResourceParametersJson(undefined, 'auth', authResourceName);
+    const frontendAuthConfig = getFrontendConfig(authParameters);
+
+    amplifyMeta.auth[authResourceName].frontendAuthConfig = amplifyMeta.auth[authResourceName].frontendAuthConfig ?? {};
+    Object.keys(frontendAuthConfig).forEach(function (key) {
+      if (!amplifyMeta.auth[authResourceName].frontendAuthConfig.hasOwnProperty(key)) {
```

Comment:
You should be able to cache `amplifyMeta.auth[authResourceName].frontendAuthConfig` so you don't have to keep computing it in a loop.

Reply:
updated